java-frontend/src/main/java/org/sonar/java/se/xproc/MethodYield.java

```diff
@@ -167,17 +162,27 @@ public boolean equals(Object obj) {
     }
     MethodYield other = (MethodYield) obj;
     return new EqualsBuilder()
-      .append(parametersConstraints, other.parametersConstraints)
+      .append(parametersConstraints.stream().flatMap(MethodYield::pmapToStream).collect(Collectors.toList()),
```

Comment:
It seems to me that it won't work. For a method as the following:
```
Object foo(Object a, Object b, Object c) {
  if (a == null && (b == null || c == null)) {
    return null;
  }
  return new Object();
}
```
We should generate the following yields (only focusing on parameters):
```
[[NULL], [NULL], [ ]]
[[NULL], [ ], [NULL]]
```
But equality tests will say that yields are equals, when they are absolutely not, as based on `flatMap` producing a new list. From how I see it, both yields are going to produce the lists [NULL, NULL].